simple_app/app_simple.py

```diff
@@ -1,46 +1,66 @@
-app_simple.vars['query'] = ''
-app_simple.vars['search'] = ''
-app_simple.vars['specid'] = ''
-app_simple.vars['source_id'] = ''
-app_simple.config['SECRET_KEY'] = os.urandom(32)
+app_simple.config['SECRET_KEY'] = os.urandom(32)  # need to generate csrf token as basic security for Flask
+
+
+def sysargs():
+    """
+    These are the system arguments given after calling this python script
 
-db_file = 'sqlite:///../SIMPLE.db'
-pd.set_option('max_colwidth', None)  # deprecation warning
+    Returns
+    -------
+    _args
+        The different argument parameters, can be grabbed via their long names (e.g. _args.host)
+    """
+    _args = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter)
+    _args.add_argument('-i', '--host', default='127.0.0.1', help='Local IP Address to host server')
+    _args.add_argument('-p', '--port', default=8000, help='Local port number to host server through', type=int)
+    _args.add_argument('-d', '--debug', help='Run Flask in debug mode?', default=False, action='store_true')
+    _args.add_argument('-f', '--file', help='Database file path relative to current directory', default='SIMPLE.db')
+    _args = _args.parse_args()
+    return _args
 
 
 class SimpleDB(Database):  # this keeps pycharm happy about unresolved references
-    Sources = None
+    Sources = None  # initialise class attribute
 
 
 class CheckResultsLength(object):
     def __call__(self, form, field):
-        db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})
-        results: pd.DataFrame = db.search_object(field.data, fmt='pandas')
-        if not len(results):
-            field.errors[:] = []
-            raise StopValidation(field.gettext('No results'))
+        db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})  # open database
+        results: pd.DataFrame = db.search_object(field.data, fmt='pandas')  # search by what is currently in searchbar
+        if not len(results):  # if that search is empty
+            field.errors[:] = []  # clear existing errors
+            raise StopValidation(field.gettext('No results'))  # stop validating and return error
 
 
 class SearchForm(FlaskForm):
-    search = StringField('', [DataRequired(), CheckResultsLength()], id='autocomplete')
-    submit = SubmitField('Query')
+    search = StringField('', [DataRequired(), CheckResultsLength()], id='autocomplete')  # searchbar
+    submit = SubmitField('Query')  # clicker button to send request
 
 
-_db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})
-allresults = _db.search_object('', fmt='pandas').source.values.tolist()
-del _db
+def all_sources():
+    db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})  # open database
+    # FIXME: db.search_object(..fmt='pandas') not returning column names
+    allresults = np.array(db.search_object(''))[:, 0].tolist()  # list all the sources by main name
```

Comment:
You may not need to do this, a query of the form db.query(db.Sources).table() may be enough to get you what you need rather than using db.search_object()